keel-core/src/main/kotlin/com/netflix/spinnaker/keel/core/api/ArtifactSummary.kt

```diff
@@ -51,8 +51,9 @@ data class ArtifactSummaryInEnvironment(
   val replacedBy: String? = null,
   val pinned: ActionMetadata? = null,
   val vetoed: ActionMetadata? = null,
-  val statefulConstraints: List<StatefulConstraintSummary> = emptyList(),
-  val statelessConstraints: List<StatelessConstraintSummary> = emptyList(),
+  val statefulConstraints: List<StatefulConstraintSummary> = emptyList(), //todo eb: remove in favor of [constraints]
+  val statelessConstraints: List<StatelessConstraintSummary> = emptyList(), //todo eb: remove in favor of [constraints]
+  val constraints: List<StatefulConstraintSummary> = emptyList(),
```

Comment:
I agree. I was actually thinking that maybe we ought to decouple the base class names from any notion of persisted state since they all have persisted state now. Something that conveys that one type is always re-evaluated on-the-fly, while the other uses stored information to determine the result, but I couldn't come up with any names today (tired 🧠).

Reply:
Ya, I was going to do clean up once this rolls out and Rani starts using it, because I'll need to go back and clean up the API anyways.